src/store/indexeddb-remote-backend.js

```diff
@@ -62,7 +64,7 @@ RemoteIndexedDBStoreBackend.prototype = {
      * @return {Promise} Resolved when the database is cleared.
      */
     clearDatabase: function() {
-        return this._doCmd('clearDatabase');
+        return this._startPromise.then(() => this._doCmd('clearDatabase'));
```

Comment:
Will it not create another backend object in https://github.com/matrix-org/matrix-js-sdk/blob/master/src/store/indexeddb-store-worker.js#L59 though?